src/MIDebugEngine/AD7.Impl/AD7Disassembly.cs

```diff
@@ -134,10 +134,16 @@ public int Seek(enum_SEEK_START dwSeekStart, IDebugCodeContext2 pCodeContext, ul
             if (iInstructions != 0)
             {
                 IEnumerable<DisasmInstruction> instructions = null;
-                _engine.DebuggedProcess.WorkerThread.RunOperation(async () =>
+                // When failing to read all instructions try looking for smaller and smaller numbers of instructions
+                // so that fetching disassembly works near the start and end of valid address spaces
+                while (instructions == null && (iInstructions > 1 || -iInstructions > 1)) 
```

Comment:
This would be a bit more efficient if the while loop was in the delegate instead of here. Otherwise it will need to thread transition back and forth.